File: api_server/app/schemas/encoding.py
"""Encoding (Base64 / UTF-8) DTOs."""

from __future__ import annotations

from pydantic import BaseModel


class EncodeRequest(BaseModel):
    """Request to Base64-encode a UTF-8 source string."""

    data: str


class EncodeResponse(BaseModel):
    """Base64 output using the RFC 4648 standard alphabet and `=` padding."""

    encoded: str


class DecodeRequest(BaseModel):
    """Request to decode an RFC 4648 Base64 string."""

    encoded: str


class DecodeResponse(BaseModel):
    """Decoded bytes re-encoded as standard Base64 to avoid binary JSON text."""

    data: str

File: frontend/src/api/encoding.ts
import client, { type APIResponse } from "./client";

export async function base64Encode(data: string): Promise<APIResponse<any>> {
  return client.post("/encoding/base64/encode", { data });
}

export async function base64Decode(encoded: string): Promise<APIResponse<any>> {
  return client.post("/encoding/base64/decode", { encoded });
}
